packages/resource/src/index.ts

```diff
@@ -298,7 +298,8 @@ export function createDeepSignal<T>(v?: T): Signal<T> {
   return [
     () => store[0],
     (update: T) => (
-      setStore(0, reconcile(typeof update === "function" ? update(unwrap(store[0])) : v)), store[0]
+      setStore(0, reconcile(typeof update === "function" ? update(unwrap(store[0])) : update)),
+      store[0]
```

Comment:
Is this line necessary? @atk

Reply:
The setter function is supposed to get the original object, not a Proxy. If it gets mutated, we need reconcile in order to facilitate the updates safely. Using the setter directly could work, but also introduce errors.